client/packages/coldchain/src/Monitoring/ListView/TemperatureChart/TemperatureChart.tsx

```diff
@@ -75,117 +85,109 @@ export const TemperatureChart = () => {
-                      <li
-                        key={`${entry.id}_legend`}
-                        style={{
-                          display: 'block',
-                          marginRight: 10,
-                          borderWidth: 0,
-                          borderBottomWidth: 1,
-                          borderTopWidth: index === 0 ? 1 : 0,
-                          borderStyle: 'solid',
-                          borderColor: theme.palette.gray.light,
-                          padding: 3,
-                        }}
-                      >
-                        <svg
-                          width="14"
-                          height="14"
-                          viewBox="0 0 32 32"
-                          style={{
-                            display: 'inline-block',
-                            verticalAlign: 'middle',
-                            marginRight: 4,
-                          }}
-                        >
-                          <path
-                            stroke="none"
-                            fill={entry.color}
-                            d="M0,8h32v12h-32z"
-                          ></path>
-                        </svg>
-                        <span>{entry.value}</span>
-                      </li>
-                    ))}
-                  </ul>
-                )}
-                payload={sensors.map(sensor => ({
-                  value: sensor.name,
-                  type: 'rect',
-                  id: sensor.id,
-                  color: sensor.colour,
-                }))}
-              />
-              <Area
-                data={breachConfig.hot}
-                type="monotone"
-                dataKey="temperature"
-                stroke={theme.palette.chart.hot.main}
-                fill={theme.palette.chart.hot.light}
-                baseValue="dataMax"
-              />
-              <Area
-                data={breachConfig.cold}
-                type="monotone"
-                dataKey="temperature"
-                stroke={theme.palette.chart.cold.main}
-                fill={theme.palette.chart.cold.light}
-              />
-              {sensors.map(sensor => (
-                <Line
-                  data={sensor.logs}
-                  key={sensor.id}
-                  dataKey="temperature"
-                  stroke={sensor.colour}
-                  type="monotone"
-                  dot={({ key, ...rest }) => (
-                    <TemperatureLineDot {...rest} key={`${sensor.id}_${key}`} />
-                  )}
-                  strokeWidth={4}
-                />
-              ))}
-            </ComposedChart>
-          </ResponsiveContainer>
-        ) : (
-          <Typography width={450}>{t('error.no-data')}</Typography>
-        )}
-      </Box>
+  if (isLoading) {
+    return <BasicSpinner />;
+  }
+
+  if (!hasData) {
+    return <NothingHere body={t('error.no-temperature-logs')} />;
```

Comment:
looks nicer, rather than having plain text on the page

Reply:
Also a big fin on early return rather then inline condition (where possible)